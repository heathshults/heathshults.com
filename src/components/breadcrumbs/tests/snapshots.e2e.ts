import { snapIt } from '../../../test';

const component = 'breadcrumbs';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('nothing is rendered with no crumbs', '<hs-breadcrumbs></hs-breadcrumbs>');
    snap(
      'one crumb',
      `<hs-breadcrumbs>
      <hs-breadcrumb>Home</hs-breadcrumb>
    </hs-breadcrumbs>`
    );
    snap(
      'two crumbs with a separator',
      `<hs-breadcrumbs>
      <hs-breadcrumb>Home</hs-breadcrumb>
      <hs-breadcrumb>Sub-section</hs-breadcrumb>
    </hs-breadcrumbs>`
    );
    snap(
      'linked crumb',
      `<hs-breadcrumbs>
      <hs-breadcrumb href="home.html">Home</hs-breadcrumb>
      <hs-breadcrumb href="another-page.html">Another page</hs-breadcrumb>
      <hs-breadcrumb>Sub-section</hs-breadcrumb>
    </hs-breadcrumbs>`
    );
  });
});
