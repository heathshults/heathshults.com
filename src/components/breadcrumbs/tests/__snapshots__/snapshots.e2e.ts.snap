// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbs renders linked crumb 1`] = `"<hs-breadcrumbs class=\\"hydrated\\"><nav><ol class=\\"hs-breadcrumbs\\"><hs-breadcrumb href=\\"home.html\\" class=\\"hydrated\\"><li class=\\"hs-breadcrumbs__crumb\\"><a class=\\"hs-link\\" href=\\"home.html\\">Home</a></li></hs-breadcrumb><hs-breadcrumb href=\\"another-page.html\\" class=\\"hydrated\\"><li class=\\"hs-breadcrumbs__crumb\\"><a class=\\"hs-link\\" href=\\"another-page.html\\">Another page</a></li></hs-breadcrumb><hs-breadcrumb class=\\"hydrated\\"><li class=\\"hs-breadcrumbs__crumb\\">Sub-section</li></hs-breadcrumb></ol></nav></hs-breadcrumbs>"`;

exports[`breadcrumbs renders nothing is rendered with no crumbs 1`] = `"<hs-breadcrumbs class=\\"hydrated\\"><nav><ol class=\\"hs-breadcrumbs\\"></ol></nav></hs-breadcrumbs>"`;

exports[`breadcrumbs renders one crumb 1`] = `"<hs-breadcrumbs class=\\"hydrated\\"><nav><ol class=\\"hs-breadcrumbs\\"><hs-breadcrumb class=\\"hydrated\\"><li class=\\"hs-breadcrumbs__crumb\\">Home</li></hs-breadcrumb></ol></nav></hs-breadcrumbs>"`;

exports[`breadcrumbs renders two crumbs with a separator 1`] = `"<hs-breadcrumbs class=\\"hydrated\\"><nav><ol class=\\"hs-breadcrumbs\\"><hs-breadcrumb class=\\"hydrated\\"><li class=\\"hs-breadcrumbs__crumb\\">Home</li></hs-breadcrumb><hs-breadcrumb class=\\"hydrated\\"><li class=\\"hs-breadcrumbs__crumb\\">Sub-section</li></hs-breadcrumb></ol></nav></hs-breadcrumbs>"`;
